src/pybind/mgr/dashboard/frontend/src/app/shared/api/cephfs.service.spec.ts

```diff
@@ -1,20 +1,85 @@
-import { HttpClientModule } from '@angular/common/http';
-import { inject, TestBed } from '@angular/core/testing';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
+import 'rxjs/add/observable/of';
+import { Observable } from 'rxjs/Observable';
 
 import { CephfsService } from './cephfs.service';
 
 describe('CephfsService', () => {
+  let service: CephfsService;
+  let httpClient: HttpClient;
+  let count = 0;
+
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpClientModule],
+      imports: [HttpClientTestingModule],
       providers: [CephfsService]
     });
+
+    service = TestBed.get(CephfsService);
+    httpClient = TestBed.get(HttpClient);
+    count = 0;
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
   });
 
   it(
-    'should be created',
-    inject([CephfsService], (service: CephfsService) => {
-      expect(service).toBeTruthy();
+    'should call list',
+    fakeAsync(() => {
```

Comment:
Sorry that I've to say this but you are only testing in this file if the method names exist. I can remove all the content of the service functions and the tests will still pass.

Reply:
This is also checking if the method returns the correct object type and if the http method is actually called.
If you remove the http call inside the function this test will fail, since we are expecting to call http get once.

Reply:
Please use HttpTestingController